R/prepare_inter_emissions_profile.R

```diff
@@ -13,6 +13,5 @@ prepare_inter_emissions_profile <- function(ep_prod, europages_companies, ecoinv
   ep_product_europages |>
     left_join(prepared_match_mapper, by = join_by_shared_cols_quietly) |>
     left_join(isic, by = "isic_4digit") |>
-    add_avg_matching_certainty("completion") |>
-    exclude_rows("risk_category")
```

Comment:
@kalashsinghal, can we safely delete this line? Removing it did not break any test so I can't be sure why is there, and whether it's dead code or a line we forgot to test.

Reply:
`exclude_rows` function is used to remove any extra rows per company which has NA in column `risk_category`. However, this function makes sure that we atleast have one row with NA `risk_category`. I don't know why the test fails. Please analyze this function [here](https://github.com/2DegreesInvesting/tiltIndicatorAfter/blob/49619c58ac51d6ea8c9b121d14555ac66d4f5383/R/utils.R#L28).

Reply:
I think this function is useful if we still want to delete these extra rows with null risk category. However, if there is a new requirement to keep the extra null rows then feel free to delete this function.

Reply:
I did some research and I can't find records of the issue that led to `exclude_rows()`. 

I see [you added that function about 8 months  ago](https://github.com/2DegreesInvesting/tiltIndicatorAfter/blame/49619c58ac51d6ea8c9b121d14555ac66d4f5383/R/utils.R#L28), but the [related PR](https://github.com/2DegreesInvesting/tiltIndicatorAfter/pull/12) doesn't link to an issue with details about what problems it solves. That change was before the [first entry in the changelog](https://2degreesinvesting.github.io/tiltIndicatorAfter/news/index.html#tiltindicatorafter-0009001-2023-09-22) and  filtering for [issues around the date of that change](https://github.com/2DegreesInvesting/tiltIndicatorAfter/issues?q=updated%3A%3C2023-07-20) is also unhelpful.

But even without detailed information, it seems that removing `exclude_rows("risk_category")` is the most conservative option. That removal would add "unmatched' products and maybe also add some rows with `NA` in `risk_category` that we don't want. But then we can remove the unwanted rows from the output. 

On the contrary, leaving `exclude_rows("risk_category")` drops "unmatched" products (which we now want), which we can't recover from the output.

So from my perspective with limited information I would go ahead and merge this PR. But I'll leave it up to you and @AnneSchoenauer to decide. Anne knows what she wants and you know tiltIndicatorAfter best.

Reply:
Hi @maurolepore. I think I have even less information than you have but I would follow your advise and go ahead... We want to have the NAs included so if this is the only option I guess it is best to delete this line.

Reply:
@maurolepore I agree that you merge this PR for now. Nevertheless, we will create a separate issue if we want to remove any extra rows in the future. :)

Reply:
OK, thanks for your comments. I'll merge then re-create the outputs and attache them to [tiltOutput-v0.0.0.9002](https://github.com/2DegreesInvesting/tiltOutput/releases/tag/v0.0.0.9002).